### ISSUE TITLE
GitHub Desktop keeps losing the current branch and offering to "Publish branch" instead

### ISSUE THREAD

**Issue by joelriendeau** [desktop/desktop#10736]
### Describe the bug

When interacting with the program, especially after an alt-tab, the program will think that my current branch does not exist anymore, and offer to publish the branch instead. Restarting sometimes fixes the issue. Waiting for 1 minute or two also fixes the issue sometimes.

### Version & OS

2.5.6, Windows 10

### Steps to reproduce the behavior

1. Go to Github Desktop after alt-tabing out
2. Program automatically loses the current branch after maybe 1 second

### Expected behavior

Should stay on Master

### Actual behavior

Instead, loses the current branch, ability to push changes. Offers to publish the branch instead.

### Screenshots

Add screenshots to help explain your problem, if applicable.

### Logs

[2020-10-05.desktop.production.log](https://github.com/desktop/desktop/files/5331047/2020-10-05.desktop.production.log)
[2020-10-06.desktop.production.log](https://github.com/desktop/desktop/files/5331048/2020-10-06.desktop.production.log)

**Comment by steveward** [comment]
Thanks for the report @joelriendeau. Is this happening in one specific repository, or in multiple repositories?

**Comment by joelriendeau** [comment]
Hi Steve, I only work in one at the moment, so the one.

**Comment by joelriendeau** [comment]
Well, I just switched repositories and the bug is definitely in one
specific repo.

**Comment by joelriendeau** [comment]
Hey Steve, did that, the issue reoccurs. I have that same repo on another machine and it works fine. I have also fully deleted the repo from my bugged machine, and re-cloned it, only to have the issue again. in an earlier attempt at fixing this.

Is it possible I need to clear the cache and the repo at the same time?

I can tell you that the Visual Studio Team Explorer git client seems to work fine. I was about to check with gitkraken.

**Comment by steveward** [comment]
@joelriendeau that's strange that a fresh copy would behave the same. Clearing out the cached data and a fresh clone usually resolves any local repo quirks. Make sure the repository isn't open in any other applications as that could prevent things from actually getting deleted as well.

**Comment by joelriendeau** [comment]
Hey again Steve, I just figured out something about my issue. I had both
the Github app and Visual Studio open at the same time. Github was in the
buggy mode. Then I closed Visual Studio, and Github immediately went back
to good mode.

So I think there might be a bad interaction between Visual Studio's git
plugin in Team Explorer, which I also use from time to time. I'll disable
it for the time being.
